<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <author>Massimo Callegari</author>
 <comment>
  Q Light Controller Plus
  inputprofileeditor.ui

  Copyright (c) 2015 Massimo Callegari

  Licensed under the Apache License, Version 2.0 (the &quot;License&quot;);
  you may not use this file except in compliance with the License.
  You may obtain a copy of the License at

      http://www.apache.org/licenses/LICENSE-2.0.txt

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an &quot;AS IS&quot; BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the License.
 </comment>
 <class>InputProfileEditor</class>
 <widget class="QDialog" name="InputProfileEditor">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>520</width>
    <height>476</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Input Profile Editor</string>
  </property>
  <property name="windowIcon">
   <iconset resource="qlcui.qrc">
    <normaloff>:/input_output.png</normaloff>:/input_output.png</iconset>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QTabWidget" name="m_tab">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="tab">
      <attribute name="title">
       <string>General</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout">
       <item row="0" column="0">
        <widget class="QLabel" name="m_manufacturerLabel">
         <property name="text">
          <string>Manufacturer</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="m_modelLabel">
         <property name="text">
          <string>Model</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="m_typeLabel">
         <property name="text">
          <string>Type</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QComboBox" name="m_typeCombo"/>
       </item>
       <item row="0" column="1">
        <widget class="QLineEdit" name="m_manufacturerEdit">
         <property name="toolTip">
          <string>The name of the company that made the device</string>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QLineEdit" name="m_modelEdit">
         <property name="toolTip">
          <string>The device's model name</string>
         </property>
        </widget>
       </item>
       <item row="3" column="0" colspan="2">
        <widget class="QGroupBox" name="m_midiGroupSettings">
         <property name="title">
          <string>MIDI Global Settings</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_3">
          <item row="0" column="0">
           <widget class="QCheckBox" name="m_noteOffCheck">
            <property name="text">
             <string>When MIDI notes are used, send a Note Off when value is 0</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="4" column="0" colspan="2">
        <spacer name="verticalSpacer_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_2">
      <attribute name="title">
       <string>Channels</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="6" column="0">
        <widget class="QGroupBox" name="m_behaviourBox">
         <property name="title">
          <string>Behaviour</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_5">
          <item row="0" column="0">
           <widget class="QLabel" name="m_movementLabel">
            <property name="text">
             <string>Movement</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QComboBox" name="m_movementCombo">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <item>
             <property name="text">
              <string>Absolute</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Relative</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QLabel" name="m_sensitivityLabel">
            <property name="text">
             <string>Sensitivity</string>
            </property>
           </widget>
          </item>
          <item row="0" column="3">
           <widget class="QSpinBox" name="m_sensitivitySpin">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimum">
             <number>10</number>
            </property>
            <property name="maximum">
             <number>100</number>
            </property>
            <property name="value">
             <number>20</number>
            </property>
           </widget>
          </item>
          <item row="1" column="0" colspan="4">
           <widget class="QCheckBox" name="m_extraPressCheck">
            <property name="text">
             <string>Generate an extra Press/Release when toggled</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="5" column="1">
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="1" column="1">
        <widget class="QToolButton" name="m_removeButton">
         <property name="toolTip">
          <string>Remove the selected channels</string>
         </property>
         <property name="text">
          <string notr="true">...</string>
         </property>
         <property name="icon">
          <iconset resource="qlcui.qrc">
           <normaloff>:/edit_remove.png</normaloff>:/edit_remove.png</iconset>
         </property>
         <property name="iconSize">
          <size>
           <width>32</width>
           <height>32</height>
          </size>
         </property>
        </widget>
       </item>
       <item row="4" column="1">
        <widget class="QToolButton" name="m_wizardButton">
         <property name="toolTip">
          <string>Automatically add channels to the list when you wiggle the device's controls</string>
         </property>
         <property name="text">
          <string notr="true">...</string>
         </property>
         <property name="icon">
          <iconset resource="qlcui.qrc">
           <normaloff>:/wizard.png</normaloff>:/wizard.png</iconset>
         </property>
         <property name="iconSize">
          <size>
           <width>32</width>
           <height>32</height>
          </size>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QToolButton" name="m_editButton">
         <property name="toolTip">
          <string>Edit the selected channel</string>
         </property>
         <property name="text">
          <string notr="true">...</string>
         </property>
         <property name="icon">
          <iconset resource="qlcui.qrc">
           <normaloff>:/edit.png</normaloff>:/edit.png</iconset>
         </property>
         <property name="iconSize">
          <size>
           <width>32</width>
           <height>32</height>
          </size>
         </property>
        </widget>
       </item>
       <item row="0" column="0" rowspan="6">
        <widget class="QTreeWidget" name="m_tree">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="MinimumExpanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="selectionMode">
          <enum>QAbstractItemView::ExtendedSelection</enum>
         </property>
         <property name="rootIsDecorated">
          <bool>false</bool>
         </property>
         <property name="itemsExpandable">
          <bool>false</bool>
         </property>
         <property name="sortingEnabled">
          <bool>true</bool>
         </property>
         <property name="allColumnsShowFocus">
          <bool>true</bool>
         </property>
         <column>
          <property name="text">
           <string>Channel</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Name</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Type</string>
          </property>
         </column>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="Line" name="line">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QToolButton" name="m_addButton">
         <property name="toolTip">
          <string>Add a new channel description</string>
         </property>
         <property name="text">
          <string notr="true">...</string>
         </property>
         <property name="icon">
          <iconset resource="qlcui.qrc">
           <normaloff>:/edit_add.png</normaloff>:/edit_add.png</iconset>
         </property>
         <property name="iconSize">
          <size>
           <width>32</width>
           <height>32</height>
          </size>
         </property>
        </widget>
       </item>
       <item row="7" column="0">
        <widget class="QGroupBox" name="m_feedbackGroup">
         <property name="title">
          <string>Custom feedback</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_4">
          <property name="topMargin">
           <number>3</number>
          </property>
          <property name="bottomMargin">
           <number>3</number>
          </property>
          <item row="0" column="2">
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>Upper value</string>
            </property>
           </widget>
          </item>
          <item row="0" column="3">
           <widget class="QSpinBox" name="m_upperSpin">
            <property name="minimum">
             <number>0</number>
            </property>
            <property name="maximum">
             <number>255</number>
            </property>
            <property name="value">
             <number>255</number>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="label">
            <property name="text">
             <string>Lower value</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QSpinBox" name="m_lowerSpin">
            <property name="maximum">
             <number>255</number>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="m_buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="qlcui.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>m_buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>InputProfileEditor</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>InputProfileEditor</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
